modules/codemod/lib/v7/updateModelSignatures.ts

```diff
@@ -0,0 +1,112 @@
+              property.key.type === 'Identifier'
+            ) {
+              const propertyName = property.key.name;
+              // Both types have a `params`, but are at different places in the AST
+              const params =
+                property.type === 'ObjectMethod'
+                  ? property.params
+                  : property.value.type === 'ArrowFunctionExpression'
+                  ? property.value.params
+                  : [];
+
+              // create a mapping for param names of callbacks and guards. This makes the logic more
+              // difficult to follow, but decreases duplication
+              const paramMapping = {
+                on: ['data', 'state'],
+                should: ['data', 'prevState'],
+              };
+
+              // We will be overriding the params with this variable later
+              const finalParams: (Identifier | ObjectPattern)[] = [];
+
+              // loop over each `paramMapping` type. `on*` and `should*` functions have different signatures
+              keys(paramMapping).forEach(key => {
+                // matches `onShow` and `shouldShow`
+                if (new RegExp(`${key}[A-Za-z]+`).test(propertyName)) {
+                  params.forEach(param => {
+                    if (param.type === 'ObjectPattern') {
+                      param.properties.forEach(property => {
+                        if (
+                          property.type === 'ObjectProperty' &&
+                          property.key.type === 'Identifier'
+                        ) {
+                          if (
+                            property.value.type === 'Identifier' ||
+                            property.value.type === 'ObjectPattern'
+                          ) {
+                            const index = paramMapping[key].indexOf(property.key.name);
+                            // use `property.value` instead of `property.key.name` in case they
+                            // renamed it example `shouldShow({ data: myData })`. The key name is
+                            // `data` while the value is `myData`
+                            finalParams[index] = property.value;
+                          }
+                        }
+                      });
+                    }
+                  });
+
+                  // `ObjectMethod` and `ObjectProperty` have params in a different place in the
+                  // AST, so apply `finalParams` where appropriate
+                  if (property.type === 'ObjectMethod') {
+                    property.params = finalParams;
+                  } else if (
+                    property.type === 'ObjectProperty' &&
+                    property.value.type === 'ArrowFunctionExpression'
+                  ) {
+                    property.value.params = finalParams;
+                  }
+                }
+              });
+            }
+          });
+        }
+      });
+    }
+  });
+
+  return root.toSource();
```

Comment:
don't forget to add this to the `index` file :)

Reply:
Done